app/controllers/api/v0/enterprises_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Api::V0::EnterprisesController < Api::V0::BaseController
+
+  def index
+    @scope = collection.ransack(params[:q]).result
+    render_collection @scope, each_serializer: Api::V0::EnterpriseSerializer, root: 'enterprises'
+  end
+
+  def show
+    @object = collection.find(params[:id])
+    render json: @object, serializer: Api::V0::EnterpriseSerializer
+  end
+
+  private
+
+  # @see Admin::EnterprisesController#collection
+  def collection
+    OpenFoodNetwork::Permissions.new(current_api_user).
+      visible_enterprises.
+      includes(:address => [:country, :state]).
+      order('is_primary_producer ASC, name')
```

Comment:
The website allows browsing all enterprises, right? (Or perhaps only those who are _visible_). The API should show the same things (but only allow updating to those the user has access to, of course); i.e. `visible_enterprises` may be too restrictive for reading.